### PR TITLE
[tbb] remove cross building restriction

### PR DESCRIPTION
**Pull request by planetmarshall**
Specify library name and version:  **tbb/\***

Remove the cross building restriction

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
cross compilation was disable because of this error: https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/7795/2-configs/macos-m1-clang/tbb/2020.1//summary.json
Do you think this was fixed ?

**Comment by ericLemanissier** [comment]
so instead of disabling all the cross-compilations, you could disable only macos cross compilation to M1. Or dig deeper and try to understand why it fails.

**Comment by planetmarshall** [comment]
I'd like to update my TBB dependencies to use the latest package from oneTBB but unfortunately that causes me a whole host of problems downstream due to this - https://github.com/conan-io/conan-center-index/issues/8308

**Comment by planetmarshall** [comment]
> so instead of disabling all the cross-compilations, you could disable only macos cross compilation to M1. Or dig deeper and try to understand why it fails.

Hmm. Mysterious... not sure it's worth digging deeper as there are newer versions of TBB available now. Seems to be just a Macos issue so have disabled it for now.

File: recipes/tbb/all/conanfile.py [hunk 1]
```diff
@@ -41,14 +41,14 @@ def config_options(self):
...
+
     def validate(self):
-        if hasattr(self, "settings_build") and tools.cross_building(self):
-            raise ConanInvalidConfiguration("Cross building is not yet supported. Contributions are welcome")
-        if self.settings.os == "Macos" and \
-           self.settings.compiler == "apple-clang" and \
-           tools.Version(self.settings.compiler.version) < "8.0":
-            raise ConanInvalidConfiguration("%s %s couldn't be built by apple-clang < 8.0" % (self.name, self.version))
+        if self.settings.os == "Macos":
+            if tools.cross_building(self):
```
Comment (ericLemanissier) [inline recipes/tbb/all/conanfile.py:47]:
```suggestion
            if hasattr(self, "settings_build") and tools.cross_building(self):
```

Comment (ericLemanissier) [reply recipes/tbb/all/conanfile.py:47]: you can see in the status checks of this PR `[required] tbb/2019_u9@ macOS, Clang — All green! (0)`, your changed disabled all macos builds, even native compilation. It should be fixed by testing self.settings_build

**Review by ericLemanissier — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**